src/esthri/src/ops/sync.rs

```diff
@@ -47,38 +64,16 @@ where
     #[cfg(not(feature = "compression"))]
     let compressed = false;
 
-    let mut glob_excludes: Vec<Pattern> = vec![];
-    let mut glob_includes: Vec<Pattern> = vec![];
-
-    if let Some(excludes) = excludes {
-        for exclude in excludes {
-            let exclude = exclude.as_ref();
-            match Pattern::new(exclude) {
-                Err(e) => {
-                    return Err(Error::GlobPatternError(e));
-                }
-                Ok(p) => {
-                    glob_excludes.push(p);
-                }
-            }
-        }
-    }
-
-    if let Some(includes) = includes {
-        for include in includes {
-            let include = include.as_ref();
-            match Pattern::new(include) {
-                Err(e) => {
-                    return Err(Error::GlobPatternError(e));
-                }
-                Ok(p) => {
-                    glob_includes.push(p);
-                }
+    let filters: Vec<GlobFilter> = match glob_filters {
+        Some(filters) => {
+            let mut filters = filters.to_vec();
+            if filters.iter().any(|x| matches!(x, GlobFilter::Include(_))) {
```

Comment:
If I'm reading this right it says "if there's any include in the list, also push an include of `*`" -- wouldn't that cause it to always include everything?

Reply:
Yep, you're right - I noticed that right after I pushed as well. Was supposed to be

```suggestion
            if !filters.iter().any(|x| matches!(x, GlobFilter::Include(_))) {
```
but I missed the negation. :man_facepalming: Currently fixing and writing a test for it!

Reply:
👍